streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingJobProgressListener.scala

```diff
@@ -101,6 +101,7 @@ private[streaming] class StreamingJobProgressListener(ssc: StreamingContext)
     runningBatchUIData(batchStarted.batchInfo.batchTime) = BatchUIData(batchStarted.batchInfo)
     waitingBatchUIData.remove(batchStarted.batchInfo.batchTime)
 
+    // TODO. this should be fixed when input stream is not receiver based stream.
```

Comment:
OK, I will keep as it is for now and remove all these TODOs. When enable this in `KafkaDirectInputDStream`, we could rethink about what exact semantics should be in this part.